storage/operation/protocol_kv_store.go

```diff
@@ -23,10 +29,32 @@ func RetrieveProtocolKVStore(r storage.Reader, protocolKVStoreID flow.Identifier
 
 // IndexProtocolKVStore indexes a protocol KV store by block ID.
 // Error returns:
-//   - storage.ErrAlreadyExists if the key already exists in the database.
+// - [storage.ErrDataMismatch] if a _different_ KV store for the given stateID has already been persisted
 //   - generic error in case of unexpected failure from the database layer
```

Comment:
>  generic error in case of unexpected failure from the database layer

For brevity, I am suggesting to remove this statement. It could be added to nearly any of our methods. The coding convention states that only expected sentinel errors need to be documented. The possibility of returning arbitrary exceptions almost always exist and is implied by the wording "Expected error returns during _normal operations_"

```suggestion
// Expected error returns during normal operations:
// - [storage.ErrDataMismatch] if a _different_ KV store for the given stateID has already been persisted
```